pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -376,6 +414,25 @@ func (p *K8sPlugin) servicesStateUpdate() error {
 	return nil
 }
 
+func (p *K8sPlugin) updateSriovService() error {
+	if p.updateTarget.sriovScript {
+		err := p.serviceManager.EnableService(p.sriovService)
+		if err != nil {
+			return err
+		}
+	}
+
+	if p.updateTarget.sriovScript {
+		err := ioutil.WriteFile(path.Join(chroot, p.sriovService.Path),
+			[]byte(p.switchdevBeforeNMRunScript.Contents.Inline), 0755)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
This function is not really clean. Why check twice? The first one still calls systemd directly

Reply:
you are right